distribution/lib/Standard/Visualization/0.0.0-dev/src/Table/Visualization.enso

```diff
@@ -211,9 +211,101 @@ make_json_for_table dataframe max_rows all_rows_count include_index_col is_db_ta
+    column = table.at column_index
+    values = column.to_vector.distinct
+    result = JS_Object.from_pairs [["distinct_vals", values]]
+    result.to_text
+
+## PRIVATE
+
+   applies the sort model to the table.
+
+   Arguments:
+   - x: table to sort
+   - sort_col_index_list: indexes of columns to sort
+   - sort_direction_list: list of sort directions 1 for ascending -1 for descending
+
+apply_sort_to_table : Table -> List -> List -> Table
+apply_sort_to_table table sort_col_index_list sort_direction_list =
+    parse_direction d = if d == -1 then Sort_Direction.Descending else Sort_Direction.Ascending
+    get_pattern i = (Sort_Column.Index (sort_col_index_list.get i) (parse_direction (sort_direction_list.get i)))
+    sort_pattern = ((0.up_to sort_col_index_list.length).map idx-> get_pattern idx)
+    table.sort sort_pattern
+
+## PRIVATE
+
+   applies one filter to the table.
+
+   Arguments:
+   - table: table to sort
+   - filter_col: the column index to filter
+   - filter_action: filter action
+   - filter_val: values to filter
+   - filter_to_val: if a Between is the filer action this is the other value to provide a range
+
+apply_single_filter_to_table : Table -> Text -> Filter_Condition -> Any -> Any -> Table
+apply_single_filter_to_table table filter_col filter_action filter_val filter_to_val = 
+    case filter_action of
+      Filter_Condition.Between _ _ _ -> table.filter (filter_col+1) (Filter_Condition.Between filter_val filter_to_val)
+      Filter_Condition.Is_In _ _ -> if filter_val.length > 0 then table.filter (filter_col+1) (Filter_Condition.Is_In filter_val) else table.filter filter_col (Filter_Condition.Is_In [""])
+      _ -> table.filter (filter_col+1) (filter_action (filter_val)) 
+
+## PRIVATE
+
+   applies the filter model to the table.
+
+   Arguments:
+   - table: table to sort
+   - filter_col: indexes of columns with sorts applies
+   - filter_action: list of filter actions 
+   - filter_val: values to filter
+
+apply_filter_to_table : Table -> Integer -> List -> List -> List -> List -> Table
+apply_filter_to_table table i filter_col filter_action filter_val filter_to_val = 
+    if i == filter_col.length then table else
+        col = filter_col.get i
+        action = filter_action.get i
+        val = filter_val.get i
+        to_val = filter_to_val.get i
+        filtered_table = apply_single_filter_to_table table col action val to_val
+        @Tail_Call apply_filter_to_table filtered_table i+1 filter_col filter_action filter_val filter_to_val
+
+## PRIVATE
+
+   Prepares the row data table for visualizations using the server side row model.
+
+   Arguments:
+   - start_number: the start index of the next 100 rows to get
+
+get_rows_for_table : Any -> Integer -> List -> List -> List -> List -> List -> List -> JS_Object
+get_rows_for_table table start_number sort_col_index_list=Nothing sort_direction_list=Nothing filter_col=Nothing filter_action=Nothing filter_val=Nothing filter_extra_val=Nothing =
```

Comment:
Are you sure these are passed in as `List`?

`List` (linked-list) is actually a type that is used relatively rarely in Enso, most cases (e.g. interop) will treat arrays as `Vector` instead.

I'd suggest trying making the signature checked and seeing if it works - IMO it will fail with Type Error expected List got Vector.

Moreover, the default `=Nothing` is also not handled. The type should be `List | Nothing` or `Vector | Nothing` (depending on the above) for the arguments that are optional.

I think it's good to ensure that the type signature (which serves as documentation) is aligned with what is actually going into these arguments.